src/ruby/yast/scr.rb

```diff
@@ -100,16 +108,24 @@ def self.UnregisterAllAgents
     #
     # It sends to component result() which force to terminate component.
     # If there is any lazy write, then it is properly finished.
-    # @param path[Yast::Path] path to agent
+    # @param path[Yast::Path, String] path to agent
     def self.UnmountAgent(path)
-      call_builtin_wrapper("UnmountAgent", path)
+      real_path = ensure_path(path)
+      call_builtin_wrapper("UnmountAgent", real_path)
     end
 
     # @private wrapper to C bindings
-    def self.call_builtin_wrapper(*args)
+    private_class_method def self.call_builtin_wrapper(*args)
       # caller(1) is one of the functions above
       res = caller(2, 1).first.match(BACKTRACE_REGEXP)
       call_builtin(res[1], res[2].to_i, *args)
     end
+
+    private_class_method def self.ensure_path(path)
+      return Path.new(path) if path.is_a?(::String)
+      return path if path.is_a?(Yast::Path)
+
+      raise ArgumentError, "argument '#{path.inspect}' is not path or string"
```

Comment:
I think it is better without the quotes, `inspect` will do a good job already. Plus exact classes, plus grammar.
```ruby
raise "Argument #{path.inspect} is neither a ::String or a Yast::Path"
```